spec/requests/api/v3/providers/courses/index_spec.rb

```diff
@@ -51,187 +52,447 @@ def perform_request
-                     "bursary_amount" => "6000",
-                     "early_career_payments" => nil,
-                     "scholarship" => nil,
-                   },
-                  },
-                  {
-                   "id" => "5",
-                   "type" => "subjects",
-                   "attributes" => {
-                     "subject_name" => "Primary with modern languages",
-                     "subject_code" => "04",
-                     "bursary_amount" => nil,
-                     "early_career_payments" => nil,
-                     "scholarship" => nil,
-                   },
-                  },
-                  {
-                   "id" => "6",
-                   "type" => "subjects",
-                   "attributes" => {
-                     "subject_name" => "Primary with physical education",
-                     "subject_code" => "06",
-                     "bursary_amount" => nil,
-                     "early_career_payments" => nil,
-                     "scholarship" => nil,
-                   },
-                  },
-                  {
-                   "id" => "7",
-                   "type" => "subjects",
-                   "attributes" => {
-                     "subject_name" => "Primary with science",
-                     "subject_code" => "07",
-                     "bursary_amount" => nil,
-                     "early_career_payments" => nil,
-                     "scholarship" => nil,
-},
-                 },
-                ],
-                "modern_languages" => nil,
+                  ],
+                  "modern_languages" => nil,
+                },
               },
+            }],
+            "links"  => {
+              "last" => "/api/v3/recruitment_cycles/2020/providers/courses?page%5Bpage%5D=1",
+            },
+            "jsonapi" => {
+              "version" => "1.0",
             },
-          }],
-          "jsonapi" => {
-            "version" => "1.0",
-          },
-        )
+          )
+        end
+
+        context "pagination" do
+          before do
+            30.times do
+              create(:course, :resulting_in_pgce_with_qts, :with_apprenticeship,
+                     level: "primary",
+                     name: "Mathematics",
+                     provider: provider,
+                     start_date: Time.now.utc,
+                     study_mode: :full_time,
+                     subjects: subjects,
+                     is_send: true,
+                     site_statuses: [courses_site_status],
+                     enrichments: [enrichment],
+                     maths: :must_have_qualification_at_application_time,
+                     english: :must_have_qualification_at_application_time,
+                     science: :must_have_qualification_at_application_time,
+                     age_range_in_years: "3_to_7",
+                     applications_open_from: applications_open_from)
+            end
+          end
```

Comment:
This is going to really slow things down. Can we break this up into contexts:

- we have one test that checks that the default `per_page` is set to 10, only requiring us to create 11 courses
- we have another test that operates with `per_page` set to 2, only requiring us to create 6 courses

Reply:
can we break this into its own file or something?

Reply:
This is all regarding the same thing, I don't think it's necessary to move it to a separate file.